src/VesselConnectButton.tsx

```diff
@@ -7,14 +7,19 @@ type VesselConnectButtonProps = {
   children?: React.ReactNode;
   style?: React.CSSProperties;
   className?: string;
+  getLinkToken: () => Promise<string>; // Retrieves and returns a linkToken
 } & ClientConfig;
 
 export default function VesselConnectButton({
   children = 'Connect CRM',
   style,
   className,
+  getLinkToken,
```

Comment:
this is by far the jankiest addition to the `useVesselLink` interface as a whole. I'm not super in love with this but it's a necessary evil to ensure the much more common case of just using `useVesselLink` works correctly.

I'll pre-empt the question of "why not just pass the `linkToken` directly", the issue is that if you load the page but don't actually open to modal for a while you'll wind up in a stale linkToken when you go to open the modal. 

We actually have this issue already but I found this PR as a great opportunity for fixing that. I'd be open to debating changing this to `linkToken` because we already have this issue now tho....